src/gensxg.c

```diff
@@ -468,25 +476,24 @@ void write_sxg(const sxg_signer_list_t* signers,
     exit(EXIT_FAILURE);
   }
 
-  FILE* out;
   if (strcmp(output, "-") == 0) {
-    out = freopen(NULL, "wb", stdout);
-    if (out == NULL) {
+    freopen(NULL, "wb", stdout);
+    if (stdout == NULL) {
       perror("reopen stdout");
       exit(EXIT_FAILURE);
     }
+    write_buffer(&result, stdout);
   } else {
-    out = fopen(output, "wb");
+    FILE* out = fopen(output, "wb");
     if (out == NULL) {
       perror("open file");
       exit(EXIT_FAILURE);
     }
-  }
-
-  size_t written = fwrite(result.data, sizeof(uint8_t), result.size, out);
-  if (written != result.size) {
-    perror("fwrite");
-    exit(EXIT_FAILURE);
+    write_buffer(&result, out);
+    if (fclose(out) != 0) {
```

Comment:
You could avoid the duplicated call to `write_buffer` if you compare `if (out != stdout && fclose(out)) {}`.

Reply:
OK, I'll adopt.